app/screens/channel/channel_base.js

```diff
@@ -111,27 +119,26 @@ export default class ChannelBase extends PureComponent {
         this.props.actions.getChannelStats(this.props.currentChannelId);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.theme !== nextProps.theme) {
+    static getDerivedStateFromProps(props, state) {
+        if (state.theme !== props.theme) {
             const options = {
                 layout: {
-                    backgroundColor: nextProps.theme.centerChannelBg,
+                    backgroundColor: props.theme.centerChannelBg,
                 },
             };
-            mergeNavigationOptions(this.props.componentId, options);
+            mergeNavigationOptions(state.componentId, options);
         }
 
-        if (nextProps.currentTeamId && this.props.currentTeamId !== nextProps.currentTeamId) {
-            this.loadChannels(nextProps.currentTeamId);
+        if (props.currentTeamId && state.currentTeamId !== props.currentTeamId) {
+            this.loadChannels(props.currentTeamId);
         }
 
-        if (nextProps.currentChannelId !== this.props.currentChannelId &&
-            nextProps.currentTeamId === this.props.currentTeamId) {
-            PushNotifications.clearChannelNotifications(nextProps.currentChannelId);
+        if (props.currentChannelId !== state.currentChannelId && props.currentTeamId === state.currentTeamId) {
+            PushNotifications.clearChannelNotifications(props.currentChannelId);
         }
 
-        if (nextProps.currentChannelId !== this.props.currentChannelId) {
-            this.props.actions.getChannelStats(nextProps.currentChannelId);
+        if (props.currentChannelId !== state.currentChannelId) {
+            state.actions.getChannelStats(props.currentChannelId);
```

Comment:
Please revisit the use of `getDerivedStateFromProps`.  It should be for state that depends on props change over time.  No side-effect; if needed, then other lifecycle methods maybe used like `componentDidUpdate`.